dandi/girder.py

```diff
@@ -556,6 +556,71 @@ def lock_dandiset(self, dandiset_identifier: str):
                         f"Failed to unlock dandiset {dandiset_identifier} due to: {msg}"
                     )
 
+    NGINX_MAX_CHUNK_SIZE = 100 * (1 << 20)
+
+    def _uploadContents(self, uploadObj, stream, size, progressCallback=None):
+        """
+        Uploads contents of a file.  Overridden so that the chunk size can be
+        set on a per-file basis.
+
+        :param uploadObj: The upload object contain the upload id.
+        :type uploadObj: dict
+        :param stream: Readable stream object.
+        :type stream: file-like
+        :param size: The length of the file. This must be exactly equal to the
+            total number of bytes that will be read from ``stream``, otherwise
+            the upload will fail.
+        :type size: str
+        :param progressCallback: If passed, will be called after each chunk
+            with progress information. It passes a single positional argument
+            to the callable which is a dict of information about progress.
+        :type progressCallback: callable
+        """
+        offset = 0
+        uploadId = uploadObj["_id"]
+
+        chunk_size = max(1 << 26, size // 1000)
```

Comment:
why not to use original super-class's chunk size instead of hardcoded `1 << 26` here?

shouldn't it be `math.ceil(size/1000)` so we do not end up with 1001 chunk?

Reply:
* I misread the original assignment and thought that `1 << 26 != MAX_CHUNK_SIZE`.  Fixed now.

* Done (but without getting floats involved).